backend/v3/storage/database/events_testing/settings_instance_test.go

```diff
@@ -0,0 +1,1790 @@
+//go:build integration
+
+package events_test
+
+import (
+	"bytes"
+	_ "embed"
+	"testing"
+	"time"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	durationpb "google.golang.org/protobuf/types/known/durationpb"
+
+	"github.com/zitadel/zitadel/backend/v3/domain"
+	"github.com/zitadel/zitadel/backend/v3/storage/database"
+	"github.com/zitadel/zitadel/backend/v3/storage/database/repository"
+	"github.com/zitadel/zitadel/internal/integration"
+	"github.com/zitadel/zitadel/pkg/grpc/admin"
+	instance "github.com/zitadel/zitadel/pkg/grpc/instance/v2beta"
+	"github.com/zitadel/zitadel/pkg/grpc/management"
+	"github.com/zitadel/zitadel/pkg/grpc/policy"
+)
+
+//go:embed picture.png
+var picture []byte
+
+//go:embed font.otf
+var font []byte
+
+func TestServer_TestInstanceLoginSettingsReduces(t *testing.T) {
+	settingsRepo := repository.LoginRepository()
+
+	t.Run("test adding login settings reduces", func(t *testing.T) {
+		ctx := t.Context()
+		before := time.Now()
+		newInstance := integration.NewInstance(t.Context())
+		after := time.Now()
+
+		retryDuration, tick := integration.WaitForAndTickWithMaxDuration(CTX, time.Second*20)
+		assert.EventuallyWithT(t, func(t *assert.CollectT) {
+			setting, err := settingsRepo.Get(
+				ctx, pool,
+				database.WithCondition(
+					database.And(
+						settingsRepo.InstanceIDCondition(newInstance.ID()),
+						settingsRepo.OrganizationIDCondition(nil),
+						settingsRepo.TypeCondition(domain.SettingTypeLogin),
+						settingsRepo.OwnerTypeCondition(domain.OwnerTypeInstance),
+					),
+				),
+			)
+			require.NoError(t, err)
+
+			// event instance.policy.login.added
+			// these values are found in default.yaml
+			assert.Equal(t, domain.OwnerTypeInstance, setting.OwnerType)
+			assert.Equal(t, true, setting.AllowRegister)
```

Comment:
Why don't you just define a settings-struct as private global and then assert the content with assert.ExportedValues?